packages/pubsub/src/subscription.js

```diff
@@ -20,50 +20,53 @@
 
 'use strict';
 
-var arrify = require('arrify');
 var common = require('@google-cloud/common');
-var commonGrpc = require('@google-cloud/common-grpc');
 var events = require('events');
+var extend = require('extend');
 var is = require('is');
-var modelo = require('modelo');
-var prop = require('propprop');
-var uuid = require('uuid');
+var os = require('os');
+var util = require('util');
 
 /**
- * @type {module:pubsub/iam}
+ * @type {module:pubsub/connectionPool}
  * @private
  */
-var IAM = require('./iam.js');
+var ConnectionPool = require('./connection-pool.js');
 
 /**
- * @type {module:pubsub/snapshot}
+ * @type {module:pubsub/histogram}
  * @private
  */
-var Snapshot = require('./snapshot.js');
+var Histogram = require('./histogram.js');
 
 /**
- * @const {number} - The amount of time a subscription pull HTTP connection to
- *     Pub/Sub stays open.
+ * @type {module:pubsub/iam}
  * @private
  */
-var PUBSUB_API_TIMEOUT = 90000;
+var IAM = require('./iam.js');
+
+/**
+ * @type {module:pubsub/snapshot}
+ * @private
+ */
+var Snapshot = require('./snapshot.js');
 
 /*! Developer Documentation
  *
  * @param {module:pubsub} pubsub - PubSub object.
- * @param {object} options - Configuration object.
- * @param {boolean} options.autoAck - Automatically acknowledge the message once
- *     it's pulled. (default: false)
- * @param {string} options.encoding - When pulling for messages, this type is
- *     used when converting a message's data to a string. (default: 'utf-8')
- * @param {number} options.interval - Interval in milliseconds to check for new
- *     messages. (default: 10)
- * @param {string} options.name - Name of the subscription.
- * @param {number} options.maxInProgress - Maximum messages to consume
- *     simultaneously.
- * @param {number} options.timeout - Set a maximum amount of time in
- *     milliseconds on an HTTP request to pull new messages to wait for a
- *     response before the connection is broken. (default: 90000)
+ * @param {string=} name - The name of the subscription.
+ * @param {object=} options - See a
```

Comment:
I think we need to move these out into the public doc block. Also, `maxConnections` doesn't seem to be documented anywhere.